crabs/detection_tracking/train_model.py

```diff
@@ -116,7 +121,68 @@ def setup_trainer(self):
             limit_train_batches=self.limit_train_batches,
         )
 
-    def train_model(self):
+    def optuna_maximise_val_precision_recall(
+        self, trial: optuna.Trial
+    ) -> float:
+        """Objective function for Optuna.
+
+        When used with Optuna, it wil maximise precision and recall on the
+        validation set.
+
+        Parameters
+        ----------
+        trial : optuna.Trial
+            The trial to optimise.
+
+        Returns
+        -------
+        float
+            The value to maximise.
+        """
+        # Sample hyperparameters from the search space for this trial
```

Comment:
I tried to make it a bit more general but let me know thoughts, not fully convinced 🤔

Reply:
I found this is confusing and longer 😅

Reply:
It was much clearer when we know exactly what hyperparameter we have for optuna

Reply:
yes, it may be more clear but it is also somewhat "hardcoded" right? With this approach we can extend it to any float or int hyperparameter. 

I personally prefer this one but it is more important that we both find this code clear so will revert to having the hyperparameters defined explicitly. Let me know if you disagree.